homeassistant/components/qnap/sensor.py

```diff
@@ -371,7 +371,7 @@ def native_value(self):
             return used
 
         if self.entity_description.key == "memory_percent_used":
-            return used / total * 100
+            return round(used / total * 100, self.entity_description.suggested_display_precision)
```

Comment:
TYhis is not correct, this is changing the sensor state using the display value. The idea behind the display precision is that it is used for the displaying purposes, not for adjusting the state machine.

../Frenck

Reply:
Thanks for the explanation.
While I agree not messing the state machine with frontend related stuff, it makes no sense to have a calculated, percentage related memory usage or volume usage using more than 2 decimals.

More over, looking at an other integration (e.g. Transmission), that one is also using calculated, percentage related states rounded to fix 2 decimals.

Would it be acceptable if the QNAP integration were doing the same, rounding the calculated percentage to 2 decimals?

thanks, Attila

Reply:
This sensor would not have been added if there was a PR now, so we also should not change this. Like Frenck expressed, the suggested_display_precision is there to suggest a display precision. It will work for 90% of the people, and there are people who want no rounding, or want a raw as possible value for whatever reason. If we round this to 2 decimals, we are busting the functionality to be more precise.

If you still want to access this value via rest, consider using a template entity to round the value.

Reply:
Thanks @joostlek , noted.